### DISCUSSION TITLE
can't use ctrl & alt commands with zellij

### DISCUSSION THREAD

**Discussion by zophiana** [ghostty-org/ghostty#3591] [Q&A] [answered]
I would like to use zellij with ghostty but unfortunately I can't use the ctrl & alt commands of zellij.

# Steps to reproduce:
0. open ghostty
1. start `zellij`
2. try to open a floating pane via `alt-f` or open a new pane via `crtl-p` `n`
3. zellij does not "react" properly

I already disabled all ghostty keybindings but does not help. The same setup works fine with Gnome Console.

Here is a detailed video: 
[Screencast From 2024-12-27 22-51-58.webm](https://github.com/user-attachments/assets/2d199311-774f-4b5d-ab8e-ccb5f551d638)

**Comment by mitchellh** [comment]
There's been other topics that covered this. I'll leave this unswered until they're found but you should be able to find a solution.

**Comment by tstorck** [comment]
I have the same issue, no input seems to reach zellij.
Zellij is configured to unlock with alt+z. Unbinding alt+z in ghostty has no effect.
I also tried
```keybind = unconsumed:alt+z=reload_config```
which reloads the configuration (notification shown) but zellij won't react to the unlock command.

Using KDE Plasma + wayland. Alacritty and Konsole work fine with this setup.

**Reply by ibanks42** [reply]
My home setup with KDE 6 + wayland works fine, but now back at work with a GNOME + Wayland setup and it I have the same issue.

Did you resolve this?

**Reply by tstorck** [reply]
No, I have no solution. 

On my machine zellij commands don't work on my KDE Plasma 6.2.4 + wayland, but if I start a Sway session instead of Plasma, it works fine. Same user, same ghostty binary, same config.

**Comment by tstorck** [comment]
I noticed something interesting: the issue is only manifesting when `numlock` is enabled.

When I disable `numlock` on my (azerty layout) keyboard, sending commands to Zellij works fine, both in plasma and sway.

When `numlock` is enabled, no commands seem to reach Zellij. 
Also `home` and `end` keys to move the cursor the the beginning and end of the line are not working when `numlock` is enabled.

**Reply by zophiana** [reply]
Yes I can confirm, this seems to be causing the issue. Thanks! This is at least a temporary fix

**Reply by 00-kat** [reply, marked as answer]
Upstream bug fixed in https://github.com/zellij-org/zellij/pull/3725, upgrade your Zellij.

See also: https://github.com/zellij-org/zellij/issues/3592.  It's not closed, but someone says that https://github.com/zellij-org/zellij/pull/3725 fixed it for them.

I was never able to reproduce this[^1], as my Zellij had that fix already.

Funnily enough, the first two search results[^2] for "zellij numlock" are https://github.com/zellij-org/zellij/issues/3592 followed by this discussion!

![A screenshot of search results.](https://github.com/user-attachments/assets/f5684e3a-e6ae-4b47-b614-b117ad3d0501)

[^1]: It's not because I didn't have number lock enabled, as I also have number lock always enabled, which I did using a compositor option, and I don't even have a number lock key on my laptop keyboard, so I couldn't toggle it even if I wanted to.
[^2]: N.B.: don't blame yourself if you think something along the lines of "drats, I should have checked for Zellij bugs!", since I would not have done so either on the assumption that since it works elsewhere it has to be a Ghostty bug had I not know that the number lock status isn't sent unless you use the Kitty Keyboard Protocol, which GNOME Console does not support but Ghostty does support.

**Reply by zophiana** [reply]
Thank you very much! I would never have thought that Zellij has the bug

**Comment by zophiana** [comment]
Closing as it has been fixed by https://github.com/zellij-org/zellij/pull/3725
